utils/sorting.go

```diff
@@ -18,26 +18,33 @@ type Sortable[T any] interface {
 	Less(T) bool
 }
 
+func LessInt[T Sortable[T]](s T, f T) int {
+	if s.Less(f) {
+		return -1
+	}
+	return 1
```

Comment:
Similarly - I don't think this handles the case that `s` and `f` are equal correctly.